lib/wanda/results.ex

```diff
@@ -0,0 +1,24 @@
+defmodule Wanda.Results do
+  @moduledoc """
+  This module exposes functionalities to interact with the historycal log of ExecutionResults.
+  It allows to:
+   - append an item to the log
+  """
+
+  alias Wanda.Execution.{ExecutionResult, Result}
+
+  @spec append(Result.t()) :: {:ok, any()} | {:error, any()}
+  def append(
```

Comment:
i find append a bit too much revealing of the underlying implementation detail, `save_result` would be more friendly to read in some months

Reply:
Changed to `save` so it looks like `Wanda.Results.save()` instead of `Wanda.Results.save_result()`

Sounds good?

Reply:
`save_result` is more idiomatic as the Results context could be dealing with more schemas, we could also find a better name for the context itself. See for example https://hexdocs.pm/phoenix/contexts.html#starting-with-generators

Reply:
Interesting piece of doc :smile: 

`Wanda.Results.save_result/1` just sounded a bit redundant, yet since it's more idiomatic I'll keep it like that.
And let's discover if the context name needs revisiting :+1: